extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallryeGraphqlProcessor.java

```diff
@@ -0,0 +1,434 @@
+    }
+
+    @BuildStep
+    void additionalBean(BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer) {
+        additionalBeanProducer.produce(AdditionalBeanBuildItem.builder()
+                .addBeanClass(GraphQLConfig.class)
+                .addBeanClass(GraphQLProducer.class)
+                .setUnremovable().build());
+    }
+
+    @BuildStep
+    void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> serviceProvider) throws IOException {
+        // Lookup Service (We use the one from the CDI Module)
+        String lookupService = SPI_PATH + LookupService.class.getName();
+        Set<String> lookupImplementations = ServiceUtil.classNamesNamedIn(Thread.currentThread().getContextClassLoader(),
+                lookupService);
+        serviceProvider.produce(
+                new ServiceProviderBuildItem(LookupService.class.getName(), lookupImplementations.toArray(new String[0])));
+
+        // Metrics (We use the one from the CDI Module)
+        String metricsService = SPI_PATH + MetricsService.class.getName();
+        Set<String> metricsImplementations = ServiceUtil.classNamesNamedIn(Thread.currentThread().getContextClassLoader(),
+                metricsService);
+        serviceProvider.produce(
+                new ServiceProviderBuildItem(MetricsService.class.getName(), metricsImplementations.toArray(new String[0])));
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    void buildExecutionService(BuildProducer<ReflectiveClassBuildItem> reflectiveClassProducer,
+            SmallRyeGraphQLRecorder recorder,
+            CombinedIndexBuildItem combinedIndex) {
+
+        IndexView index = combinedIndex.getIndex();
+        Schema schema = SchemaBuilder.build(index);
+
+        reflectiveClassProducer
+                .produce(new ReflectiveClassBuildItem(true, true, getClassesToRegisterForReflection(schema)));
+
+        recorder.createExecutionService(schema);
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    void buildEndpoints(
+            BuildProducer<RequireBodyHandlerBuildItem> requireBodyHandlerProducer,
+            BuildProducer<RouteBuildItem> routeProducer,
+            BuildProducer<NotFoundPageDisplayableEndpointBuildItem> notFoundPageDisplayableEndpointProducer,
+            //BuildProducer<ExtensionSslNativeSupportBuildItem> extensionSslNativeSupportProducer,
+            SmallRyeGraphQLRecorder recorder,
+            ShutdownContextBuildItem shutdownContext) throws IOException {
+
+        /*
+         * <em>Ugly Hack</em>
+         * In dev mode, we pass a classloader to use in the CDI Loader.
+         * This hack is required because using the TCCL would get an outdated version - the initial one.
+         * This is because the worker thread on which the handler is called captures the TCCL at creation time
+         * and does not allow updating it.
+         *
+         * In non dev mode, the TCCL is used.
```

Comment:
This is being worked on by @stuartwdouglas IIUC the issue.

Reply:
Yes, once his changes is in I can remove this and the special QuarkusClassloadingService class